TotalCrossVM/src/nm/ui/android/skia.cpp

```diff
@@ -251,6 +252,20 @@ int skia_makeBitmap(int32 id, void *data, int32 w, int32 h) {
         SkBitmap bitmap;
 		bitmap.installPixels(SkImageInfo::Make(w, h, kN32_SkColorType, kUnpremul_SkAlphaType),
 							 (void*)converted, sizeof(Pixel) * w, releaseProc, nullptr);
+
+#if USE_COLORTYPE_CONVERSION
+        void* dstPixels = operator new(w * h * canvas->imageInfo().bytesPerPixel());
```

Comment:
The first conversion swaps ARGB to BGRA I believe, Skia doesn't support ARGB directly.
The second allocation does not necessarily have the same size of the first one, because it uses the system's pixel format, it will usually be smaller, but it may be bigger? Not sure.

This second array is used by bitmap.readPixel, which performs the conversion from BGRA to the system's pixel format (RGB888, RGB565 or whatever).
Then we install this second array into the original bitmap, along with all the updated info. The install calls the releaseProc, which frees the first array - I double checked that, we're not leaking memory here. ;)

So for now, we do need both conversions, but later I'm thinking about changing the Image object to use BGRA instead of ARGB, which would allow us to drop the first conversion.